server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -236,12 +236,17 @@ export function configureWebSocketServices(
                 return handleServerError(logger, errMsg, claims.documentId, claims.tenantId);
             }
 
+            const connectedTimestamp = Date.now();
```

Comment:
Are there any mechanisms running on the server to ensure this time is in sync with the timestamp that is added to ops? These two services may run on different machines right.

Reply:
The timestamp for ops is handled by alfred, the same one creating that connectedTimestmap. So they will be in sync

Reply:
Oh okay, great. I thought deli adds it for ops.

Reply:
But I believe that does not add any additional guarantees, as two front-ends may have time drift, and as such two clients connecting to document may assume different order RE who joined first if they use timestamps vs. order they see clients added to audience / quorum.
Maybe other way to say it - time may go backwards when looking at op stream.
Is my understanding correct here?

Reply:
Basically, does this mean different clients will see different timestamp for the same op? Which timestamp does the server use for the messages it stores?

Reply:
The frontends decide on the timestamp for the submitted ops. When deli sequences ops, it uses the timestamp sent from the frontend for the op timestamp. That is the only timestamp for ops - all clients will see that same op.
For vlads question - yes time may go backwards in the op stream if different frontends are submitting ops and are out of sync